### PR TITLE
[FIX]es in check printing

### PR DESCRIPTION
**Pull request by Whisno**
- Module account_check_writing becomes account_check_printing, a hidden dependency for modules which provide check templates
- Payments by check have a button to go back from state 'sent' to 'posted'
- Amount in words is hidden

Goes with https://github.com/odoo/enterprise-9/pull/6

Note :
- I didn't find how to hide a module. It would be nice to hide 'Check Printing Base' because it is useless by itself.
- The tests pass locally, but I am not sure how to test the PR since it's split between community and enterprise. Maybe first merging this one since runbot builds the enterprise feature branches using community master ?

### CODE REVIEW

File: addons/account/models/res_config.py [hunk 1]
```diff
@@ -69,9 +69,6 @@ class AccountConfigSettings(models.TransientModel):
     period_lock_date = fields.Date(related='company_id.period_lock_date', help="Only users with the 'Adviser' role can edit accounts prior to and inclusive of this date")
     fiscalyear_lock_date = fields.Date(string="Fiscal Year lock date", related='company_id.fiscalyear_lock_date', help="No users, including Advisers, can edit accounts prior to and inclusive of this date")
 
-    module_account_check_writing = fields.Boolean(string='Pay your suppliers by check',
```
Comment (Whisno) [inline addons/account/models/res_config.py]: Maybe l10n_us can add this ?

**Comment by qdp-odoo** [comment]
what's the reason why the test is moved into the l10n_us_printing module?

**Comment by Whisno** [comment]
because account_check_writing is not supposed to work alone and just throws an exception if no module extend do_print_checks, so it makes sense to me to test it through an implementation

**Comment by qdp-odoo** [comment]
ha yes, it makes sense..

**Comment by qdp-odoo** [comment]
* account_check_printing must be installable alone, but at that time it has to be transparent (new payment method 'check' but no button to print check)
* make l10n_us dependant of account_check_printing
* l10n_us_check_printing has to be auto_install true and dependant of l10n_us

**Comment by Whisno** [comment]
Ok, so account_check_printing has xml views to add the 'print check' button and the 'x checks to print' on the dashboard. But they should be executed when l10n_us_check_printing or another implementation is installed. Any idea how to do this ?

I found a solution that is ... hum ... that just works. In the 'data' key of the __openerp__.py of l10n_us_check_printing :
'../../odoo/addons/account_check_printing/views/account_journal_dashboard_view.xml',

Otherwise, I was thinking about something like <function model="ir.ui.view" name="_enable_check_printing"/> that would execute the views from account_check_printing when l10n_us_check_printing is installed. But that would probably end up with me wasting work hours having fun with the framework.

**Pull request merged**